src/posix/platform/netif.cpp

```diff
@@ -159,14 +438,31 @@ static void UpdateMulticast(otInstance *aInstance, const otIp6Address &aAddress,
 
     assert(sInstance == aInstance);
 
-    VerifyOrExit(sIpFd > 0, OT_NOOP);
+    VerifyOrExit(sIpFd >= 0, OT_NOOP);
     memcpy(&mreq.ipv6mr_multiaddr, &aAddress, sizeof(mreq.ipv6mr_multiaddr));
     mreq.ipv6mr_interface = sTunIndex;
 
-    VerifyOrExit(
-        setsockopt(sIpFd, IPPROTO_IPV6, (aIsAdded ? IPV6_JOIN_GROUP : IPV6_LEAVE_GROUP), &mreq, sizeof(mreq)) == 0,
-        perror("setsockopt");
-        error = OT_ERROR_FAILED);
+    int err;
+    err = setsockopt(sIpFd, IPPROTO_IPV6, (aIsAdded ? IPV6_JOIN_GROUP : IPV6_LEAVE_GROUP), &mreq, sizeof(mreq));
+    if ((err != 0) && (errno == EINVAL) && (IN6_IS_ADDR_MC_LINKLOCAL(&mreq.ipv6mr_multiaddr)))
```

Comment:
We already have that -- it's two lines below.  But if I don't squelch this error case on Mac and FreeBSD, the code will never proceed.  And because it dies before bringing up the interface, it will actually die every time.

I'm happy to wrap this special handing for Mac and FreeBSD (and eventually, I'd like to remove the workaround).